include/crypto/ecx.h

```diff
@@ -33,15 +33,15 @@
 #  define X25519_BITS           253
 #  define X25519_SECURITY_BITS  128
 
-#  define X448_BITS             448
+#  define X448_BITS             448  // ?? but the group order is 446 bits
```

Comment:
// Is not good for some compiler...
@mattcaswell might be the best person to answer your question..

Reply:
I *think* this is the key size (56 bytes == 448)

Reply:
okay.   the values for `X448_BITS`, `ED25519_BITS`, `ED448_BITS` make sense if they are meant to be the bit-length of the secrets/private keys.  But, the value for `X25519_BITS` is 253 -- that does not fit that pattern.  A raw X25519 secret is 32 bytes.

In RFC 7748, there is the following paragraph:

>  Scalars are assumed to be randomly generated bytes.  For X25519,
in order to decode 32 random bytes as an integer scalar, set the
three least significant bits of the first byte and the most
significant bit of the last to zero, set the second most
significant bit of the last byte to 1 and, finally, decode as
little-endian.  This means that the resulting integer is of the
form 2^254 plus eight times a value between 0 and 2^251 -
1 (inclusive).  Likewise, for X448, set the two least significant
bits of the first byte to 0, and the most significant bit of the
last byte to 1.  This means that the resulting integer is of the
form 2^447 plus four times a value between 0 and 2^445 -
1 (inclusive).

Maybe someone set `X25519_BITS` to 253 because of this paragraph.

in any case, maybe this discussion needs to be moved elsewhere.  I can open an issue for it if you think it needs one.

Reply:
seems related to #19070

Reply:
Yeah...looks like there is some inconsistency here.